src/Microsoft.DocAsCode.Build.Engine/SystemMetadataGenerator.cs

```diff
@@ -48,24 +48,26 @@ public SystemMetadata Generate(InternalManifestItem item)
             var fileWithoutWorkingFolder = file.RemoveWorkingFolder();
             attrs.Path = fileWithoutWorkingFolder;
 
-            // 1. Root Toc is always in the top directory of output folder
-            var rootToc = _toc.FirstOrDefault();
-            if (rootToc != null)
+            // 1. Root Toc is specified by RootTocKey, or by default in the top directory of output folder
+            if (!string.IsNullOrEmpty(_context.RootTocPath))
             {
-                var rootTocPath = rootToc.File.RemoveWorkingFolder();
-                if (rootTocPath.SubdirectoryCount == 0)
+                if (((TypeForwardedToRelativePath)_context.RootTocPath).IsFromWorkingFolder())
```

Comment:
Seems no need. removed.

---
In reply to: [89045892](https://github.com/dotnet/docfx/pull/908#discussion_r89045892) [](ancestors = 89045892)